<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>444</width>
    <height>530</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>444</width>
    <height>530</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>444</width>
    <height>530</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Calculator</string>
  </property>
  <property name="styleSheet">
   <string notr="true">#MainWindow{
background-color: rgb(40, 40, 40);
border-radius: 20px;
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="maximumSize">
    <size>
     <width>16777215</width>
     <height>508</height>
    </size>
   </property>
   <property name="styleSheet">
    <string notr="true">#centralwidget {
border: 1px solid rgb(0, 0, 0);
border-radius: 5px;
background-color: rgb(40,  40, 40); }
</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <property name="bottomMargin">
     <number>9</number>
    </property>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_6">
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QFrame" name="formFrame1">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="styleSheet">
         <string notr="true">#formFrame1{
background-color:  rgb(40, 40, 40);
border: 1px solid rgb(0, 0, 0);
border-radius: 5px;
}</string>
        </property>
        <layout class="QFormLayout" name="formLayout_5">
         <property name="topMargin">
          <number>20</number>
         </property>
         <property name="bottomMargin">
          <number>50</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="answerTextLabel">
           <property name="styleSheet">
            <string notr="true">#answerTextLabel{color: white; }
</string>
           </property>
           <property name="text">
            <string>Answer:                                                      </string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="answerLabel">
           <property name="styleSheet">
            <string notr="true">#answerLabel{color: white; }
</string>
           </property>
           <property name="text">
            <string>0</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QFormLayout" name="formLayout">
        <property name="bottomMargin">
         <number>2</number>
        </property>
        <item row="0" column="0" colspan="2">
         <widget class="QLineEdit" name="userInputEntry">
          <property name="styleSheet">
           <string notr="true">#userInputEntry{
background-color: rgb(40, 40, 40);
color: white;
border: 1px solid rgb(0, 0, 0);
border-radius: 5px;
}</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLineEdit" name="inputEntry">
        <property name="styleSheet">
         <string notr="true">#inputEntry{
border: 1px solid rgb(0, 0, 0);
border-radius: 5px;
background-color: white;
color: black;}
</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignBottom|Qt::AlignRight|Qt::AlignTrailing</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QFrame" name="formFrame">
      <property name="styleSheet">
       <string notr="true">#formFrame {
border: 1px solid rgb(0, 0, 0);
border-radius: 5px;
background-color: rgb(40, 40, 40); }
</string>
      </property>
      <layout class="QFormLayout" name="formLayout_4">
       <property name="bottomMargin">
        <number>1</number>
       </property>
       <item row="5" column="0">
        <widget class="QLabel" name="hexLabel">
         <property name="styleSheet">
          <string notr="true">#hexLabel{
color: white;
}</string>
         </property>
         <property name="text">
          <string>Hex: </string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QLabel" name="hexResult">
         <property name="styleSheet">
          <string notr="true">#hexResult{
color: white;
}</string>
         </property>
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="decLabel">
         <property name="styleSheet">
          <string notr="true">#decLabel{
color: white;
}</string>
         </property>
         <property name="text">
          <string>Dec: </string>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <widget class="QLabel" name="decResult">
         <property name="styleSheet">
          <string notr="true">#decResult{
color: white;
}</string>
         </property>
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QLabel" name="octLabel">
         <property name="styleSheet">
          <string notr="true">#octLabel{
color: white;
}</string>
         </property>
         <property name="text">
          <string>Oct: </string>
         </property>
        </widget>
       </item>
       <item row="7" column="1">
        <widget class="QLabel" name="octResult">
         <property name="styleSheet">
          <string notr="true">#octResult{
color: white;
}</string>
         </property>
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
       <item row="8" column="0">
        <widget class="QLabel" name="binLabel">
         <property name="styleSheet">
          <string notr="true">#binLabel{
color: white;
}</string>
         </property>
         <property name="text">
          <string>Bin: </string>
         </property>
        </widget>
       </item>
       <item row="8" column="1">
        <widget class="QLabel" name="binResult">
         <property name="styleSheet">
          <string notr="true">#binResult{
color: white;
}</string>
         </property>
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_23">
      <property name="topMargin">
       <number>10</number>
      </property>
      <item>
       <widget class="QPushButton" name="bitORButton">
        <property name="styleSheet">
         <string notr="true">#bitORButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>bit OR</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="bitANDButton">
        <property name="styleSheet">
         <string notr="true">#bitANDButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>bit AND</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="bitXORButton">
        <property name="styleSheet">
         <string notr="true">#bitXORButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>bit XOR</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="leftShiftButton">
        <property name="styleSheet">
         <string notr="true">#leftShiftButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>&lt;&lt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="rightShiftButton">
        <property name="styleSheet">
         <string notr="true">#rightShiftButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>&gt;&gt;</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_24">
      <item>
       <widget class="QPushButton" name="exponentButton">
        <property name="styleSheet">
         <string notr="true">#exponentButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>x^</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="squareButton">
        <property name="styleSheet">
         <string notr="true">#squareButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>^2</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="rootButton">
        <property name="styleSheet">
         <string notr="true">#rootButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>x√</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="factorialButton">
        <property name="styleSheet">
         <string notr="true">#factorialButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>!</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="complexNumberButton">
        <property name="styleSheet">
         <string notr="true">#complexNumberButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>j</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_25">
      <item>
       <widget class="QPushButton" name="logButton">
        <property name="styleSheet">
         <string notr="true">#logButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>log10</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="naturalLogButton">
        <property name="styleSheet">
         <string notr="true">#naturalLogButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>ln</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="inverseSinButton">
        <property name="styleSheet">
         <string notr="true">#inverseSinButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>sin-1</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="inverseCosButton">
        <property name="styleSheet">
         <string notr="true">#inverseCosButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>cos-1</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="inverseTanButton">
        <property name="styleSheet">
         <string notr="true">#inverseTanButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>tan-1</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_26">
      <item>
       <widget class="QPushButton" name="openBracesButton">
        <property name="styleSheet">
         <string notr="true">#openBracesButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>(</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="closeBracesButton">
        <property name="styleSheet">
         <string notr="true">#closeBracesButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="sineButton">
        <property name="styleSheet">
         <string notr="true">#sineButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>sin</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="cosineButton">
        <property name="styleSheet">
         <string notr="true">#cosineButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>cos</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="tangentButton">
        <property name="styleSheet">
         <string notr="true">#tangentButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>tan</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_27">
      <item>
       <widget class="QPushButton" name="sevenButton">
        <property name="styleSheet">
         <string notr="true">#sevenButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>7</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="eightButton">
        <property name="styleSheet">
         <string notr="true">#eightButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>8</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="nineButton">
        <property name="styleSheet">
         <string notr="true">#nineButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>9</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="deleteButton">
        <property name="styleSheet">
         <string notr="true">#deleteButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>Del</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="resetButton">
        <property name="styleSheet">
         <string notr="true">#resetButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>AC</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_28">
      <item>
       <widget class="QPushButton" name="fourButton">
        <property name="styleSheet">
         <string notr="true">#fourButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>4</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="fiveButton">
        <property name="styleSheet">
         <string notr="true">#fiveButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>5</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="sixButton">
        <property name="styleSheet">
         <string notr="true">#sixButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>6</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="timesButton">
        <property name="styleSheet">
         <string notr="true">#timesButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>*</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="divideButton">
        <property name="styleSheet">
         <string notr="true">#divideButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>/</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_29">
      <item>
       <widget class="QPushButton" name="oneButton">
        <property name="styleSheet">
         <string notr="true">#oneButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>1</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="twoButton">
        <property name="styleSheet">
         <string notr="true">#twoButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>2</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="threeButton">
        <property name="styleSheet">
         <string notr="true">#threeButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>3</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="plusButton">
        <property name="styleSheet">
         <string notr="true">#plusButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>+</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="minusButton">
        <property name="styleSheet">
         <string notr="true">#minusButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>-</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_30">
      <item>
       <widget class="QPushButton" name="zeroButton">
        <property name="styleSheet">
         <string notr="true">#zeroButton{
color: red;
pressed: blue;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="radixPointButton">
        <property name="styleSheet">
         <string notr="true">#radixPointButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>.</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="piButton">
        <property name="styleSheet">
         <string notr="true">#piButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>π</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="answerButton">
        <property name="styleSheet">
         <string notr="true">#answerButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>Ans</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="equalToButton">
        <property name="styleSheet">
         <string notr="true">#equalToButton{
color: red;
background-color: black;}</string>
        </property>
        <property name="text">
         <string>=</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
